core/src/main/java/org/elasticsearch/index/query/FuzzyQueryParser.java

```diff
@@ -60,23 +68,23 @@ public FuzzyQueryBuilder fromXContent(QueryParseContext parseContext) throws IOE
                 if (token == XContentParser.Token.FIELD_NAME) {
                     currentFieldName = parser.currentName();
                 } else {
-                    if ("term".equals(currentFieldName)) {
+                    if (parseContext.parseFieldMatcher().match(currentFieldName, TERM_FIELD)) {
```

Comment:
Good to read, but this more or less repeats what ParseFieldMatcher.match() already does. I think it just adds another stop on the road to the actual match-implementation, which is not even in ParseFieldMatcher but in the ParseField itself. I was thinking about simple shortening like `matcher.match(currentFieldName, PARSE_FIELD)`, but I think its fine the way it is right now in the PR, not worth going trough all the files again IMHO.